src/repo/mod.rs

```diff
@@ -241,20 +255,34 @@ impl<TRepoTypes: RepoTypes> Repo<TRepoTypes> {
     }
 
     /// Remove block from the block store.
-    pub async fn remove_block(&self, cid: &Cid) -> Result<(), Error> {
+    pub async fn remove_block(&self, cid: &Cid) -> Result<Cid, Error> {
         let cid = cid.as_upgraded_cid();
         if self.is_pinned(&cid).await? {
             return Err(anyhow::anyhow!("block to remove is pinned"));
         }
+
         // sending only fails if the background task has exited
         self.events
             .clone()
             // unprovide only cidv1
             .send(RepoEvent::UnprovideBlock(cid.clone()))
             .await
             .ok();
-        self.block_store.remove(&cid).await?;
-        Ok(())
+
+        // FIXME: Need to change location of pinning logic.
+        // I like this pattern of the repo abstraction being some sort of
+        // "clearing house" for the underlying result enums, but this
+        // could potentially be pushed out out of here up to Ipfs, idk
+        match self.block_store.remove(&cid).await? {
+            Ok(success) => match success {
+                BlockRm::Removed(cid) => Ok(cid),
```

Comment:
the issue here is that all of the methods I touched while doing the cidv0->cidv1 is that if they return the cid, they return the cid the user used, not the upgraded one as is done here.

Reply:
I removed the clone from Ipfs::remove_block, which will pass the user-supplied Cid to Repo::remove_block, which upgrades it and then passes it down to the BlockStore That should be sufficient, right? We could potentially upgrade it after retrieval

Reply:
I think on this line it should be optionally downgraded, which is a bit of chore (you'd need to check version, codec and hash). I think the easiest would be to rename the original parameter as `original`, keep the other cids but return `Ok(original.clone())` here.

It does however make the `BlockRm::Removed(HERE)` <-- the HERE a bit redundant.

Reply:
Another way is to just remove the `Cid` from return value and use the original `src/lib.rs` version here: https://github.com/ipfs-rust/rust-ipfs/pull/153/files#diff-b4aea3e418ccdb71239b96952d9cddb6R337.

If the cid is not taken by value, the caller already has the original id and can continue using it.